wormhole/file_transport.go

```diff
@@ -184,40 +184,73 @@ type fileTransport struct {
 	appID           string
 }
 
-func (t *fileTransport) connectViaRelay(otherTransit *transitMsg) (net.Conn, error) {
+// combines hints and remove duplicates to minimaze connections
+func appendAndRemoveDuplicateHints(myTransit []transitHintsV1, peerTransit []transitHintsV1, hintType string) []transitHintsRelay {
+	var combinedHints []transitHintsRelay
+
+	for _, peerHints := range peerTransit {
+		if peerHints.Type == hintType {
+			combinedHints = peerHints.Hints
+
+		}
+
+	}
+
+	for _, myHints := range myTransit {
+		if myHints.Type == hintType {
+			for _, myHint := range myHints.Hints {
+				var duplicate bool = false
+				for _, peerHint := range combinedHints {
+					if myHint == peerHint {
+						duplicate = true
+					}
+				}
+				if !duplicate {
+					combinedHints = append(combinedHints, myHint)
+				}
+			}
+		}
+
+	}
+	return combinedHints
+}
+
+func (t *fileTransport) connectViaRelay(myTransit *transitMsg, peerTransit *transitMsg) (net.Conn, error) {
```

Comment:
We don't access the `transitMsg.AbilitiesV1` field here. Therefore we could just pass in the merged `[]transitHintsV1` instead. (Tbf this should already have been done before, but now is the next best time to do it)

Reply:
Not to sure if I understood. We don't need AbilitiesV1 here. Or you think we just need to pass append(myTransit, peerTransit...) to function?

Reply:
@vu3rdd, @piegamesde refactored that part, take a look :) Plus added few test cases and jump on new case, which we don't handle in w-w :(